lib/_stream_readable.js

```diff
@@ -493,7 +493,9 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
   var ondrain = pipeOnDrain(src);
   dest.on('drain', ondrain);
 
+  var cleanedUp = false;
   function cleanup() {
+    cleanedUp = true;
```

Comment:
Wouldn't it be better if we flipped the flag after the clean up activities are actually done?